backfill.sh

```diff
@@ -0,0 +1,140 @@
+#!/bin/bash
+# this script is used to update the article-json stored in lax
+# performing just an INGEST of content, *no* PUBLISH events are
+# sent whatsoever.
+
+# talking to lax via the adaptor.py script for many thousands of articles is
+# extremely slow, so this script bypasses all that, bulk generates articles,
+# bulk validates them then tells lax to do a bulk ingest.
+
+set -euo pipefail # strict mode
+#set -xv # debugging
+
+# housekeeping
+
+trap ctrl_c INT
+function ctrl_c() {
+    echo "caught ctrl-c"
+    exit 1
+}
+
+srcdir=$(pwd)
+errcho(){ >&2 echo $@; }
+
+# where articles will be linked to/downloaded for backfill
+defaultrunpath="run-$(date +'%Y%m%d%H%M%S')"
+runpath=${1:-$defaultrunpath}
+runpath=$(realpath $runpath) # realpath needs to be installed :(
```

Comment:
ah yes, some distros have it installed by default, some not. I was going to see what Ubuntu had by default (it doesn't). I think there are clever bash-isms to resolve symlinks and get the full absolute path , but they're so arcane ... I'll update the builder-base now to have it installed by default.